admin-console/src/main/webapp/src/components/SystemBehaviorGroups/BehaviorGroupTable.tsx

```diff
@@ -0,0 +1,182 @@
+import { Breadcrumb, BreadcrumbItem, Button, PageSection, Spinner,
+    Title, Toolbar, ToolbarContent, ToolbarItem } from '@patternfly/react-core';
+import { PencilAltIcon, TrashIcon } from '@patternfly/react-icons';
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import * as React from 'react';
+import { useParams } from 'react-router';
+
+import { useBundleTypes } from '../../services/Applications/GetBundleById';
+import { useCreateSystemBehaviorGroup } from '../../services/SystemBehaviorGroups/CreateSystemBehaviorGroup';
+import { useDeleteBehaviorGroup } from '../../services/SystemBehaviorGroups/DeleteSystemBehaviorGroup';
+import { useSystemBehaviorGroups } from '../../services/SystemBehaviorGroups/GetBehaviorGroups';
+import { useUpdateBehaviorGroupActionsMutation } from '../../services/SystemBehaviorGroups/UpdateActions';
+import { BehaviorGroup } from '../../types/Notifications';
+import { CreateEditBehaviorGroupModal } from './CreateEditBehaviorGroupModal';
+import { DeleteBehaviorGroupModal } from './DeleteBehaviorGroupModal';
+
+type BundlePageParams = {
+    bundleId: string;
+}
+
+export const BehaviorGroupsTable: React.FunctionComponent = () => {
+    const getBehaviorGroups = useSystemBehaviorGroups();
+    const { bundleId } = useParams<BundlePageParams>();
+    const getBundles = useBundleTypes(bundleId);
+    const newBehaviorGroup = useCreateSystemBehaviorGroup();
+    const deleteBehaviorGroupMutation = useDeleteBehaviorGroup();
+    const updateBehaviorGroupActions = useUpdateBehaviorGroupActionsMutation();
+
+    const columns = [ 'System Behavior Group', 'Actions' ];
+
+    const [ showModal, setShowModal ] = React.useState(false);
+    const [ showDeleteModal, setShowDeleteModal ] = React.useState(false);
+
+    const [ isEdit, setIsEdit ] = React.useState(false);
+    const [ isOpen, setIsOpen ] = React.useState(false);
+
+    const [ systemBehaviorGroup, setSystemBehaviorGroup ] = React.useState<Partial<BehaviorGroup>>({});
+
+    const bundle = React.useMemo(() => {
+        if (getBundles.payload?.status === 200) {
+            return getBundles.payload.value;
+        }
+
+        return undefined;
+    }, [ getBundles.payload?.status, getBundles.payload?.value ]);
+
+    const createBehaviorGroup = () => {
+        setShowModal(true);
+        setIsEdit(false);
+        setSystemBehaviorGroup({});
+    };
+
+    const editSystemBehaviorGroup = (b: BehaviorGroup) => {
+        setShowModal(true);
+        setIsEdit(true);
+        setSystemBehaviorGroup(b);
+
+    };
+
+    const deleteBehaviorGroupModal = (b: BehaviorGroup) => {
+        setShowDeleteModal(true);
+        setSystemBehaviorGroup(b);
+    };
+
+    const handleUpdate = React.useCallback((systemBehaviorGroup) => {
```

Comment:
I don't feel confident in this doing what I think it should be doing. I would need to add the updateActions to the rest, but I can't quite put the pieces together.

Reply:
Could you explain ?